### ISSUE TITLE
HalfCheetahDirEnv meta-evaluation

### ISSUE THREAD

**Issue by bordeauxred** [rlworkgroup/garage#1365]
Hi there, 

I tried to replicate the pearl results from the pearl paper, however for halfcheetah forward/backward (script attached as txt) I got, using the hyper parameters similar to the oyster repo https://github.com/katerakelly/oyster/blob/master/configs/cheetah-dir.json and https://github.com/katerakelly/oyster/blob/master/configs/default.py ) a strange result. 
The returns are in absolute values as expected,  however the  Cheetah is either running always forward or always backward within one evaluation phase. (so the return for forward is very high (x) and for backward very low (-x) or vice versa) 
As the directions are always sampled the sampling I tried a) to hack the sampling to include always a forward and a backward task and to shuffle the order b) to sample 8 test and train tasks to make sure each direction is at least seen once. The above described behavior of running always into the same direction within one meta-evaluation was observed in both cases. 
Do you have by chance an intuition what might cause this and how to fix it?
[pearl_half_cheetah_dir.txt](https://github.com/rlworkgroup/garage/files/4558838/pearl_half_cheetah_dir.txt)

**Comment by ryanjulian** [comment]
@bordeauxred thanks for reporting a possible bug! It's user reports like yours which really help us make the software better.

I think we need a little more information from you to help us track it down.

Can you give us a report with the following template?

# Expected Behavior

# Observed Behavior

# Steps to Reproduce
1.
2.
3.

# Possible solutions or things you've tried (and whether they worked or not)

# More info

**Comment by bordeauxred** [comment]
Expected:
I expected to see for halfcheetahdir (using the oyster hyper-parameters  https://github.com/katerakelly/oyster/blob/master/configs/cheetah-dir.json and https://github.com/katerakelly/oyster/blob/master/configs/default.py ))a learning curve as in the paper pearl paper:
![Screenshot from 2020-04-30 20-54-53](https://user-images.githubusercontent.com/28216059/80748237-e547e880-8b24-11ea-9cad-52a31505e669.png)

i instead got 
![hc_dir](https://user-images.githubusercontent.com/28216059/80751838-ccdacc80-8b2a-11ea-8519-900bcd7771fa.png)

steps to reproduce:
I can't attach .py files, thus I changed the format to .txt. Please change the type back to .py and you could run it. 
[pearl_half_cheetah_dir.txt](https://github.com/rlworkgroup/garage/files/4560858/pearl_half_cheetah_dir.txt)


Things I tried:
When using num_train_tasks = num_test_tasks = 2  the train/ test data sometimes consisting only of one direction, thus I a) sampled either 8 (which is not necessarily balanced) or b) changed the task sampling in halfcheetahdir to        
if num_tasks ==2:
   directions = [-1.,1.]
   np.random.shuffle(directions) 
in order to have a balanced test/ trainset. 

I have no glue what causes this odd behavior which occurred for me over different seeds. 

Please let me know if you need any further information.

**Comment by ryanjulian** [comment]
@bordeauxred Thanks for the detailed report.

In the MetaTest plot above, what is `num_test_tasks` set to?

**Comment by ryanjulian** [comment]
If you want to speed up debugging, you could help us by sending us the experiment directory in question (in data/local) via a file-sharing service, and/or uploading the TensorBoard outputs to https://tensorboard.dev so we can look at them for you.

**Comment by krzentner** [comment]
Thanks again for the detailed report.
Looking at the graph, I think your guess that there's something wrong with how the MetaEvaluator and PEARL are coupled together is right. I've found what might be the bug in `PEARL.adapt_policy`, and I'm running an experiment now to verify it.

**Comment by bordeauxred** [comment]
@krzentner 
Thank you, this solves the problem.
